site-src/geps/gep-917.md

```diff
@@ -0,0 +1,162 @@
+
+- Make it easier for API consumers to understand what a particular API does
+- Make it possible for tooling to be constructed to check for portability
+between implementations
+
+Additionally, as the first project to develop an "official" set of CRDs, we have
+a responsibility to the community to build out a set of best practices for
+similar efforts in the future. Ensuring that whatever we build is reusable for
+other projects will help to lift everyone who works with CRDs.
+
+### What do we need out of conformance for Gateway API?
+Must have:
+
+- Support for only implementing some of the Gateway API CRD objects. Some objects,
+like Gateway itself, are required for all implementations, but implementations
+may choose what Route objects they support. There will probably be some common
+sets of Route objects supported across similar implementations, but this proposal
+expects that we will make calls about what to call that common experience at a
+later date.
+- Support for fields or features that have Core, Extended, and
+ImplementationSpecific support. In particular, it must be possible for
+implementations to only support some subset of Extended fields, and to be able
+to use the framework for their own ImplementationSpecific features if required.
+- A testing suite that can validate that an implementation meets the conformance
+profiles it claims.
+- A way to retrieve conformance information in a machine-parseable way.
+
+## Proposal
+
+### Conformance profiles principles
+
+#### Basics
+
+The Gatway API project defines conformance purely in terms of what objects an
+implementation supports.
+By "supports" in this context, we mean the functionality covered by Core support.
+Extended support fields are handled by a separate mechanism.
+
+All implementations must support all the Core functions on the following objects:
+- GatewayClass
+- Gateway
+- ReferencePolicy
+
+The following objects are optional to support, but have defined behavior if you
+do:
+- HTTPRoute
+- TLSRoute
+- TCPRoute
+- UDPRoute
+
+For all of these objects, we should aim to have the usual range of tests for
+both the happy and unhappy paths for various types of operations.
+
+The conformance is versioned - it tracks the required features for a specific
+version of the API, and must be included in and updated by a version bump in the
+bundle version of the Gateway API. (The _bundle version_ is the thing that we
+mark as a "release", that looks like `v0.4.0`, not `v1alpha2`). 
+
+This will enable an implementation to say that it supports version whatever of
+the Gateway API. This is again similar to upstream in that implementations need
+to submit conformance test results for each version of Kubernetes they support.
+
+Because the support is defined in terms of the objects that an implementation
+supports, the conformance is composable, and orthogonal for each object type.
+It's valid to only support TLSRoute and not HTTPRoute, for example, although
+it would be odd.
```

Comment:
Tiny nit: Maybe focus on a more common example such as supporting HTTPRoute but not TCPRoute.

Reply:
Also, remove the "it would be odd", we may be potentially shooting down some cases. Others in the community will be (are?) more creative and use this API in interesting ways.